### PR TITLE
refactor(Telemetry): Rename `analytics` to `telemetry`

### PR DESCRIPTION
**Pull request by pgrzesik**
Addresses: #9257

### CODE REVIEW

File: test/unit/lib/utils/telemetry/index.test.js [hunk 1]
```diff
@@ -27,9 +27,9 @@ describe('analytics', () => {
   };
 
   before(() => {
-    process.env.SLS_ANALYTICS_URL = analyticsUrl;
-    ({ report, sendPending } = proxyquire('../../../../../lib/utils/analytics/index.js', {
-      '@serverless/utils/analytics-and-notfications-url': analyticsUrl,
+    process.env.SLS_ANALYTICS_URL = telemetryUrl;
```
Comment (medikoo) [inline test/unit/lib/utils/telemetry/index.test.js:30]: I think this env var can be removed (looks as same not cleaned leftover)

**Review by medikoo — COMMENTED** [review]
@pgrzesik Looks great, I've just proposed to remove an extra line (assuming it's fine)

File: test/unit/lib/utils/telemetry/index.test.js (hunk 1 above)
Comment (pgrzesik) [reply test/unit/lib/utils/telemetry/index.test.js]: Good catch, it's already mocked on next lines :+1:

**Review by medikoo — APPROVED** [review]
Looks great 👍

**Pull request merged**